tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -262,6 +262,22 @@ static PyObject * THPVariable_integral_scalar(PyObject* self, PyObject* args) {
   END_HANDLE_TH_ERRORS
 }
 
+static Tensor dispatch_invert(const Tensor & self) {
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(self);
+  return 1 - self;
```

Comment:
I think invert on tensors used to be a bitwise operation on all bits, not only the first one

Reply:
It's currently `1 - self`:

https://github.com/pytorch/pytorch/blob/5c809de4b4663c688ebc2cd389c2c866aa22f6e5/torch/tensor.py#L374-L377

It's a not ideal because `__invert__`/`~` is supposed to be bitwise invert, but it's useful since we mostly use `ByteTensor` as a tensor of booleans.

Reply:
That is true, but if we go with the bitwise path, and treat 0 as false, and all other values as true, it will retain its current semantics.

Reply:
That seems like a reasonable definition for `ByteTensor.__invert__`, but I'd rather not change it right now. Some other functions require only 0s or 1s as inputs. (For example, `maskedSelect` throws if mask contains values > 1).

I'd still prefer adding a true tensor of booleans instead of having ByteTensor play double duty of uint8 and bool.

Reply:
Yeah, I agree we should keep it as is for now